astroquery/heasarc/core.py

```diff
@@ -1,92 +1,246 @@
-# Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-from typing import Union
+import os
+
+import shutil
+import requests
+import tarfile
 import warnings
-from io import StringIO, BytesIO
+import numpy as np
 from astropy.table import Table
-from astropy.io import fits
 from astropy import coordinates
 from astropy import units as u
+from astropy.utils.decorators import deprecated, deprecated_renamed_argument
+
+import pyvo
+
+from astroquery import log
 from ..query import BaseQuery
-from ..utils import commons
-from ..utils import async_to_sync
+from ..utils import commons, async_to_sync, parse_coordinates
 from ..exceptions import InvalidQueryError, NoResultsWarning
 from . import conf
 
-__all__ = ['Heasarc', 'HeasarcClass']
-
-
-def Table_read(*args, **kwargs):
-    if commons.ASTROPY_LT_5_1:
-        return Table.read(*args, **kwargs)
-    else:
-        return Table.read(*args, **kwargs, unit_parse_strict='silent')
-
 
 @async_to_sync
 class HeasarcClass(BaseQuery):
+    """Class for accessing HEASARC data using XAMIN.
 
-    """
-    HEASARC query class.
+    Example Usage:
+    ...
 
-    For a list of available HEASARC mission tables, visit:
-        https://heasarc.gsfc.nasa.gov/cgi-bin/W3Browse/w3catindex.pl
     """
 
-    URL = conf.server
-    TIMEOUT = conf.timeout
-    coord_systems = ['fk5', 'fk4', 'equatorial', 'galactic']
+    # we can move url to Config later
+    VO_URL = conf.VO_URL
+    TAR_URL = conf.TAR_URL
+    S3_BUCKET = conf.S3_BUCKET
+    timeout = conf.timeout
+
+    def __init__(self):
+        """Initialize some basic useful variables"""
+        super().__init__()
+        self._tap = None
+        self._datalink = None
+        self._meta_info = None
+        self._session = None
+
+    @property
+    def tap(self):
+        """TAP service"""
+        if self._tap is None:
+            self._tap = pyvo.dal.TAPService(
+                f'{self.VO_URL}/tap', session=self._session
+            )
+            self._session = self._tap._session
+        return self._tap
+
+    @property
+    def _meta(self):
```

Comment:
This is a great idea, and great that the server returns the info in the table and not in the metadata of the table, as e.g astropy has serious toubles with parsing that non-column metadata